packit_service/worker/handlers/bodhi.py

```diff
@@ -0,0 +1,130 @@
+# Copyright Contributors to the Packit project.
+# SPDX-License-Identifier: MIT
+
+"""
+This file defines classes for job handlers related to Bodhi
+"""
+import logging
+from typing import Optional
+
+from packit.constants import DEFAULT_BODHI_NOTE
+
+from packit.exceptions import PackitException
+
+from packit.api import PackitAPI
+from packit.config.aliases import get_branches
+
+from packit.config import JobConfig, JobType, PackageConfig
+from packit.local_project import LocalProject
+from packit.utils.repo import RepositoryCache
+from packit_service.constants import CONTACTS_URL, KojiBuildState
+from packit_service.worker.events.koji import KojiBuildEvent
+from packit_service.worker.handlers.abstract import (
+    JobHandler,
+    TaskName,
+    configured_as,
+    reacts_to,
+)
+from packit_service.worker.result import TaskResults
+
+logger = logging.getLogger(__name__)
+
+
+@configured_as(job_type=JobType.bodhi_update)
+@reacts_to(event=KojiBuildEvent)
+class CreateBodhiUpdateHandler(JobHandler):
+    """
+    This handler can create a bodhi update for successful Koji builds.
+    """
+
+    topic = "org.fedoraproject.prod.buildsys.build.state.change"
+    task_name = TaskName.bodhi_update
+
+    def __init__(
+        self,
+        package_config: PackageConfig,
+        job_config: JobConfig,
+        event: dict,
+    ):
+        super().__init__(
+            package_config=package_config,
+            job_config=job_config,
+            event=event,
+        )
+
+        # lazy properties
+        self._koji_build_event: Optional[KojiBuildEvent] = None
+
+    @property
+    def koji_build_event(self):
+        if not self._koji_build_event:
+            self._koji_build_event = KojiBuildEvent.from_event_dict(
+                self.data.event_dict
+            )
+        return self._koji_build_event
+
+    def pre_check(self) -> bool:
+        """
+        We react only on finished builds (=KojiBuildState.open state)
```

Comment:
It's strange that the completed builds are marked as `OPEN` (in fedmsg payload?), but I believe you.

Reply:
maybe [this](https://fedora-fedmsg.readthedocs.io/en/latest/topics.html#buildsys-build-state-change) is the enum that should be used here? the currently used enum is for [buildsys-task-state-change](https://fedora-fedmsg.readthedocs.io/en/latest/topics.html#buildsys-task-state-change)

Reply:
Ah, I might have missed that last time when I was checking that page. Thanks!